src/components/Form.test.js

```diff
@@ -305,11 +330,209 @@ describe("Form", () => {
       screen.getByText("You can't use inappropriate words.")
     ).toBeInTheDocument();
     expect(screen.getByText("Max 500 words.")).toBeInTheDocument();
+    expect(
+      screen.getByText("Please enter a street number")
+    ).toBeInTheDocument();
   });
 
   it("with testId", () => {
     const { container } = render(<SimpleForm testId="my-form" />);
 
     expect(container.firstChild).toHaveAttribute("data-testid", "my-form");
   });
+
+  describe("calling exposed functions from render child", () => {
+    it("sets form state correctly when setErrors is called", async () => {
+      const initialValues = {
+        name: "",
+        age: "",
+        address: { streetNumber: "", streetName: "" },
+      };
+
+      const initialErrors = {
+        name: ["Please enter a name"],
+        age: ["Please enter an age"],
+      };
+
+      const RenderChild = ({ setErrors, state }) => {
+        useEffect(() => {
+          if (state.values.name === "Helena") {
+            setErrors({
+              name: [
+                "This name is already taken",
+                "Try to spell it differently",
+              ],
+              "address.streetNumber": "Please enter a street number",
```

Comment:
Thanks for testing a nested key!

Reply:
Had to as the original implementation on setErrors was't handling nested keys correctly

Reply:
Thanks a lot for fixing this!